foundations/javascript_basics/fundamentals-5.md

```diff
@@ -22,14 +22,59 @@ Objects are a _very_ important part of the JavaScript language, and while for th
 
 Besides being a quick and handy way to store data, arrays also have a set of functions for manipulating that data in very powerful ways.  Once you begin to master these functions you will start to see ways to use them all over the place! There are really only a handful of these functions... but as you'll soon see, the possibilities of what you can do with them are near endless.
 
-1. Start out by watching [this video](https://www.youtube.com/watch?v=HB1ZC7czKRs) from Wes Bos.  To follow along, use your local Javascript30 repository. If you don't have the repo yet, clone it from [here](https://github.com/wesbos/JavaScript30).
-2. Watch and code with [Part 2](https://www.youtube.com/watch?v=QNmRfyNg1lw) of Wes Bos's array series.
-3. For a more comprehensive and in-depth guide to array methods in JavaScript, you can also check out this [array method guide](https://javascript.info/array-methods). This resource covers each method in detail, with examples and explanations of their usage.
+As an example of what we mean, let's consider a `sumOfTripledEvens` function. It will:
+- Take in an array
+- For every even number, it will triple it
+- Then it will sum all those even numbers
+
+Currently, if you wanted to implement a function like that you may write something similar to this:
+
+```javascript
+function sumOfTripledEvens(array) {
+  let sum = 0;
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 === 0) {
+      sum += array[i] * 3;
+    }
+  }
+  return sum;
+}
+```
+
+We have a function `sumOfTripledEvens` that takes in an `array`, it then iterates through the array and for every even number it adds triple that number to sum. After the `for` loop is complete, it returns the sum.
```

Comment:
Just for increased clarity, let's add a note that this function introduces three problems that need to be solved. Can you think of what they are? (Possibly a spoiler tag)

1. We need to perform an operation on *only the even numbers*
1. We need to transform those numbers into triple their value
1. We need to add the results up

This can help to clarify why we might need to call three different methods to get our result.

Additionally, this can give us a chance to explain filter, map, and reduce. Filter: I need an array with some of these items, but possibly not all of them. Map: I need an array with all of these items, but the values need to be transformed in some way. Reduce: I need some data (any type) that can only be calculated with all of the values in this array.

Reply:
I've completely overhauled the section, now it goes in-depth about each method specifically and also gives a brief introduction on callbacks. What do you think about it?